airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/typing_deduping/BigqueryDestinationInitialStatusGatherer.kt

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.typing_deduping
+
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.Field
+import com.google.cloud.bigquery.QueryJobConfiguration
+import com.google.cloud.bigquery.StandardSQLTypeName
+import com.google.cloud.bigquery.StandardTableDefinition
+import com.google.cloud.bigquery.TableDefinition
+import com.google.cloud.bigquery.TableId
+import com.google.cloud.bigquery.TimePartitioning
+import com.google.common.annotations.VisibleForTesting
+import io.airbyte.cdk.load.command.Append
+import io.airbyte.cdk.load.command.Dedupe
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.command.Overwrite
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.message.Meta
+import io.airbyte.cdk.load.orchestration.ColumnNameMapping
+import io.airbyte.cdk.load.orchestration.DestinationInitialStatusGatherer
+import io.airbyte.cdk.load.orchestration.TableName
+import io.airbyte.cdk.load.orchestration.TableNames
+import io.airbyte.cdk.load.orchestration.legacy_typing_deduping.AlterTableReport
+import io.airbyte.cdk.load.orchestration.legacy_typing_deduping.FinalTableInitialStatus
+import io.airbyte.cdk.load.orchestration.legacy_typing_deduping.RawTableInitialStatus
+import io.airbyte.cdk.load.orchestration.legacy_typing_deduping.TypingDedupingDestinationInitialStatus
+import io.airbyte.cdk.util.CollectionUtils.containsAllIgnoreCase
+import io.airbyte.cdk.util.CollectionUtils.containsIgnoreCase
+import io.airbyte.cdk.util.CollectionUtils.matchingKey
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.math.BigInteger
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
+private val logger = KotlinLogging.logger {}
+
```

Comment:
moved out of the old BigqueryDestinationHandler, and updated to use the new interfaces.